docs/DeveloperGuide.md

```diff
@@ -1,18 +1,616 @@
+* `description`
+* `entryType` - Can be `Constants.EntryType.INC` or `Constants.EntryType.INC` 
+depending on whether the entry is an income or expenditure respectively.
+* `amount`
+* `start` and `end` - Which months does the entry apply to. Set to 1 and 12 by 
+default (i.e. occurs every month)
+* `isAuto` - Indicates whether the entry is automatically deducted/credited from/to account, 
+or manually deducted/credited from/to account
+* `notes` - Any user-specified notes
+
+`RecurringTrackerList` extends ItemList, and supports the following methods on top of inherited methods
+* `addItem(Item)` - Override. Adds item and sorts according to the day in ascending order
+* `getEntriesFromDayXtoY` - Returns an ArrayList of all entries that fall between day X and Y 
+(provided by developer in the code, not by user). Mainly used for reminders
+
+##### Reminders
+Upon launching the program, the system date and time is recorded in `RunHistory`.
+
+The program then checks if there are any entries upcoming within 5 days from the current date, and prints the entries out
+as reminders.
+
+1. Main code calls `MenuPrinter#printReminders()`, which in turn calls 
+`ReminderListGenerator#generateListOfRemindersAsStrings()`. 
+1. `ReminderListGenerator` checks the current date, and calculates the day of month which is 5 days from current date.
+This is stored in `dayToRemindUntil`.
+1. `ReminderListGenerator` then checks if `dayToRemindUntil` is after the last day of the current month. If it is,
+then the reminder timeframe will overflow to the next month. 
+    
+    For example:
+    * Current date is 29th October. There are 31 days in October. 5 days after today is 34th, 
+    which is beyond last day of October.
+    * Reminder timeframe will overflow to next month, until 3rd of November
+
+1. If it has overflown, set `isOverflowToNextMonth` to true. Subtract the last day of month from `dayToRemindUntil`.
+The new value of `dayToRemindUntil` is the day of next month that the reminder timeframe extends to.
+
+    For example:
+    * Continuing from example earlier, `dayToRemindUntil = 34`.
+    * `dayToRemindUntil -= NUM_DAYS_IN_OCT`, i.e. 34 - 31
+    * `dayToRemindUntil = 3`, representing that the reminder timeframe extends to 3rd of November
+1. `ReminderListGenerator` then grabs the entries within the reminder timeframe from the list of all recurring entries.
+    * If `isOverflowToNextMonth == true`, it will grab all entries from `currentDay` to `lastDayOfMonth` 
+    and all entries from `1` (1st day of next month) to `dayToRemindUntil`
+    * Else, it will simply grab all entries from `currentDay` to `dayToRemindUntil`
+
+1. Lastly, the list of entries will be converted to a formatted String to be displayed as reminders, and passed back
+to `MenuPrinter`, who will pass it to `UiManager` to print.
+
+The sequence diagram below shows how it works:
+
+![](uml_images/recurringtracker/images/reminderSeqDiagram.png)
+
+
+
+#### Feature 2: FinanceTools
```

Comment:
Perhaps this might be feature 3 instead?

Reply:
Noted, thanks for correcting!